akka-javasdk/src/main/java/akka/javasdk/agent/RemoteMcpTools.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.agent;
+
+import akka.annotation.DoNotInherit;
+import akka.http.javadsl.model.HttpHeader;
+import akka.javasdk.impl.agent.RemoteMcpToolsImpl;
+
+import java.util.function.Predicate;
+
+/**
+ * Access to tools from one remote MCP server.
+ *
+ * <p>Not for user extension, create instances using {@link #fromServer(String)}.
+ */
+@DoNotInherit
+public interface RemoteMcpTools {
+
+  /**
+   * @param serverUri A URI to the remote MCP HTTP server, for example "https://example.com/sse" or
+   *     "https://example.com/mcp
+   */
+  static RemoteMcpTools fromServer(String serverUri) {
+    return new RemoteMcpToolsImpl(serverUri);
+  }
+
+  /**
+   * Define a filter to select what discovered tool names are passed on to the chat model. Names
+   * that are filtered will not be described to the model and will not allow calls.
+   */
+  RemoteMcpTools allowToolNames(Predicate<String> toolNameFilter);
+
+  RemoteMcpTools withToolInterceptor(ToolInterceptor interceptor);
+
+  RemoteMcpTools addClientHeader(HttpHeader header);
+
+  interface ToolInterceptor {
+    // FIXME appropriate Java types here - do we parse request payload into json, user types with
+    // jackson or what?
```

Comment:
This is the biggest question mark for me: How we should present the request and response payloads to the user. 

Request payload is always JSON but it could be a bother to have to model the format, which is specific for a remote tool (at one point in time/version of the remote server) as a Java class. On the other hand, to inspect it in any meaningful way you'd probably want to parse it. Do we leave that to users to deal with or do we provide API for it.

Response payload can be plain text or whatever so even harder to have anything else than String for it.

Reply:
Yeah, we can possibly add some convenience that takes a type and tries to parse/render request payload as it, but let's hold that off until we see a clear need.